### PR TITLE
Added Tensorboard Logging callback

### PR DESCRIPTION
**Pull request by awilliamson**
Additionally added log-dir argparse.

Not sure if this is particularly wanted. But I know quite a few users of Retinanet have been falling back to some Tensorboard logging. This is rather simplistic, but can be expanded upon later for more complex behaviours.

### CODE REVIEW

File: keras_retinanet/bin/train.py [hunk 1]
```diff
@@ -113,6 +113,20 @@ def create_callbacks(model, training_model, prediction_model, validation_generat
...
+        tb = keras.callbacks.TensorBoard(
+            log_dir                 =   args.log_dir,
+            histogram_freq          =   0,
+            batch_size              =   args.batch_size,
+            write_graph             =   True,
+            write_grads             =   False,
+            write_images            =   False,
+            embeddings_freq         =   0,
+            embeddings_layer_names  =   None,
+            embeddings_metadata     =   None
```
Comment (de-vri-es) [inline keras_retinanet/bin/train.py]: The PEP8 check is unhappy about the multiple spaces after the operator. The aligning is fine (please keep it :) ), but it wants only a single space after the `=`

Comment (awilliamson) [reply keras_retinanet/bin/train.py]: Yes PEP8 hates me :) Do you want 1 space after `=`, or do you want fully PEP8 compliance with regards to the space.
E.g `write_images            = False,` or `write_images            =False,`?

Comment (de-vri-es) [reply keras_retinanet/bin/train.py]: Definitely one space, PEP8 can go cry in a corner if it doesn't like it :)

Comment (awilliamson) [reply keras_retinanet/bin/train.py]: Done :)

**Comment by hgaiser** [comment]
> I'm just not sure on the name of the flag --log-dir. I feel the name should include tensorboard somewhere. Maybe just --tensorboard or --tensorboard-dir?

I agree with @de-vri-es , `--log-dir` makes it sound like it saves log output there. Could you rename it to `--tensorboard-dir` ?

**Comment by awilliamson** [comment]
@hgaiser Done :)

File: keras_retinanet/bin/train.py [hunk 2]
```diff
@@ -234,7 +234,7 @@ def parse_args(args):
     parser.add_argument('--epochs',        help='Number of epochs to train.', type=int, default=50)
     parser.add_argument('--steps',         help='Number of steps per epoch.', type=int, default=10000)
     parser.add_argument('--snapshot-path', help='Path to store snapshots of models during training (defaults to \'./snapshots\')', default='./snapshots')
-    parser.add_argument('--log-dir',       help='Log directory for Tensorboard output', default='./logs')
+    parser.add_argument('--tensorboard-dir',       help='Log directory for Tensorboard output', default='./logs')
```
Comment (hgaiser) [inline keras_retinanet/bin/train.py]: Could you align this ? :+1:

**Pull request merged**